deepchecks/vision/datasets/classification/simple.py

```diff
@@ -0,0 +1,205 @@
+    else:
+        raise TypeError(f'Unknown value of object_type - {object_type}')
+
+
+class SimpleClassificationDataset(VisionDataset):
+    """Simple VisionDataset type for the classification tasks.
+
+    Current class expects that data within root folder
+    will be structured in a next way:
+
+        - root/
+            - train/
+                - class1/
+                    image1.jpeg
+            - test/
+                - class1/
+                    image1.jpeg
+
+    Otherwise exception will be raised.
+
+    Parameters
+    ----------
+    root : str
+        Path to the root directory of the dataset.
+    train : bool
+        if `True` train dataset, otherwise test dataset
+    transforms : Callable, optional
+        A function/transform that takes in an PIL image and returns a transformed version.
+        E.g, transforms.RandomCrop
+    transform : Callable, optional
+        A function/transforms that takes in an image and a label and returns the
+        transformed versions of both.
+        E.g, ``transforms.Rotate``
+    target_transform : Callable, optional
+        A function/transform that takes in the target and transforms it.
+    image_extension : str, default 'JPEG'
+        images format
+    """
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transforms: t.Optional[t.Callable] = None,
+        transform: t.Optional[t.Callable] = None,
+        target_transform: t.Optional[t.Callable] = None,
+        image_extension: str = 'JPEG'
+    ) -> None:
+        self.root_path = Path(root).absolute()
+
+        if not (self.root_path.exists() and self.root_path.is_dir()):
+            raise ValueError(f'{self.root_path} - path does not exist or is not a folder')
+
+        super().__init__(str(self.root_path), transforms, transform, target_transform)
+        self.image_extension = image_extension.lower()
+
+        if train is True:
+            self.images = sorted(self.root_path.glob(f'train/*/*.{self.image_extension}'))
+        else:
+            self.images = sorted(self.root_path.glob(f'test/*/*.{self.image_extension}'))
+
+        if len(self.images) == 0:
+            raise ValueError(f'{self.root_path} - is empty or has incorrect structure')
+
+        # class label -> class index
+        self.classes_map = t.cast(t.Dict[str, int], {
+            img.parent.name: index
+            for index, img in enumerate(self.images)
+        })
+        # class index -> class label
+        self.reverse_classes_map = t.cast(t.Dict[int, str],{
+            v: k
+            for k, v in self.classes_map.items()
+        })
+
+    def __getitem__(self, index: int) -> t.Tuple[pilimage.Image, int]:
+        """Get the image and label at the given index."""
+        image_file = self.images[index]
+        image = cv2.imread(str(image_file))
+        image = pilimage.fromarray(cv2.cvtColor(image, cv2.COLOR_BGR2RGB))
```

Comment:
can just return the cv2 image, why convert to PIL?

Reply:
`cv2.imread` returns a NumPy array, not some image obj instance.
yup, better to return an array.
but/ then another question is - how should we pass it to the `self.transforms`, as an array or as `PIL.Image`?
do we make any assumptions about what is used as `transformers`?

Reply:
Only check that uses the transforms is Robustness and it requires a model, so I think we don't have any special assumptions regarding the transforms. You can perhaps copy the way transforms work in the classification_tutorial.rst

Reply:
done